src/components/Filter/hooks/useMultiFilterUtils.ts

```diff
@@ -0,0 +1,114 @@
+import { isEqual } from 'lodash';
+import { useEffect, useState } from 'react';
+
+import { FilterOption } from '../types';
+import { Props as FilterProps } from '../types';
+import { SELECT_ALL_OPTION } from 'components/Select/constants';
+
+type Props = Pick<
+  FilterProps,
+  'multi' | 'selectedItems' | 'items' | 'defaultValue' | 'onClear' | 'onFilterDelete'
+> & {
+  setFilterLabel: (value: string) => void;
+  setIsOpen: (value: boolean) => void;
+  setSearchValue: (value: string) => void;
+};
+
+const useMultiFilterUtils = ({
+  multi,
+  selectedItems = [],
+  items,
+  defaultValue,
+  setFilterLabel,
+  setIsOpen,
+  onClear,
+  setSearchValue,
+  onFilterDelete = () => {},
+}: Props) => {
+  const [availableMultiFilters, setAvailableMultiFilters] = useState<FilterOption[]>([]);
+
+  const [multiFilters, setMultiFilters] = useState(selectedItems);
+
+  useEffect(() => {
+    if (multi) {
+      const selectedFilterValues = multiFilters?.map(({ value }) => value) ?? [];
+
+      setAvailableMultiFilters(
+        items?.filter((item) => !selectedFilterValues.includes(item.value)) ?? []
+      );
+    }
+  }, [items, multiFilters, multi]);
+
+  const updateMultiFilterLabel = (items: FilterOption[]) => {
+    if (items.length > 0) {
+      if (items.length === 1) {
+        setFilterLabel(items[0].label);
+      } else {
+        setFilterLabel(`${items[0].label} + ${items.length - 1} more`);
+      }
+    } else {
+      setFilterLabel(defaultValue.label);
+    }
+  };
+
+  const handleOptionDelete = (option?: FilterOption) => {
+    let newItems: FilterOption[] = [];
```

Comment:
The general consensus on using `let` is, and correct me if I'm wrong, that you should only use it when you know that a variable will change. In this case, the variable is a constant. I'd advocate for this to be turned into two `const` variables declared in each if statement, or another approach altogether.

Reply:
Also consider wrapping this function in useCallback